lib/chai/interface/assert.js

```diff
@@ -590,6 +591,56 @@ assert.isNotFunction = function (val, msg) {
   new Assertion(val, msg, assert.isNotFunction, true).to.not.be.a('function');
 };
 
+/**
+ * TODO
+ */
+assert.isCallable = function (val, msg) {
+  if (!['Function', 'AsyncFunction', 'GeneratorFunction', 'AsyncGeneratorFunction'].includes(type(val))) {
+    throw new AssertionError(
+      msg + ': expected ' + inspect(val) + ' to be callable',
+      undefined,
+      assert.isCallable
+    );
+  }
```

Comment:
I think this might be wrong and should just live in `core/assertions.js` and we can call that from here. I'll clean that up later.